programs/lz4io.c

```diff
@@ -1362,47 +1398,204 @@ const char* LZ4IO_blockTypeID(int sizeID, int blockMode, char* buffer)
+                          /* Now we've consumed frameInfo we can use it to store the total contentSize */
+                          frameInfo.lz4FrameInfo.contentSize += cfinfo->frameSummary.lz4FrameInfo.contentSize;
+                        }
+                        else{
+                          DISPLAYLEVEL(3, " %20llu %20s %9s \n", totalBlocksSize + hSize, "-", "-");
+                          cfinfo->allContentSize = 0;
+                        }
+                        result = LZ4IO_LZ4F_OK;
+                      } }
+                  }
+                } }
+            } }
+            break;
+        case LEGACY_MAGICNUMBER:
+            frameInfo.frameType = legacyFrame;
+            if (cfinfo->frameSummary.frameType != legacyFrame && cfinfo->frameCount !=0) cfinfo->eqFrameTypes = 0;
+            cfinfo->eqBlockTypes = 0;
+            cfinfo->allContentSize = 0;
+            { const unsigned long long totalBlocksSize = LZ4IO_skipLegacyBlocksData(finput);
+            if (totalBlocksSize){
+              DISPLAYLEVEL(3, "    %6llu %14s %5s %8s %20llu %20s %9s\n",
+                              cfinfo->frameCount + 1,
+                              LZ4IO_frameTypeNames[frameInfo.frameType],
+                              "-", "-",
+                              totalBlocksSize + 4,
+                              "-", "-");
+              result = LZ4IO_LZ4F_OK;
+            } }
+            break;
+        case LZ4IO_SKIPPABLE0:
+            frameInfo.frameType = skippableFrame;
+            if (cfinfo->frameSummary.frameType != skippableFrame && cfinfo->frameCount !=0) cfinfo->eqFrameTypes = 0;
+            cfinfo->eqBlockTypes = 0;
+            cfinfo->allContentSize = 0;
+            {   nbReadBytes = fread(buffer, 1, 4, finput);
+                if (nbReadBytes != 4)
+                    EXM_THROW(42, "Stream error : skippable size unreadable");
+            }
+            {   unsigned const size = LZ4IO_readLE32(buffer);
+                int const errorNb = fseek_u32(finput, size, SEEK_CUR);
+                if (errorNb != 0)
+                    EXM_THROW(43, "Stream error : cannot skip skippable area");
+              DISPLAYLEVEL(3, "    %6llu %14s %5s %8s %20u %20s %9s\n",
+                              cfinfo->frameCount + 1,
+                              "SkippableFrame",
+                              "-", "-", size + 8, "-", "-");
+
+                result = LZ4IO_LZ4F_OK;
+            }
+            break;
+        default:
+            {   long int const position = ftell(finput);  /* only works for files < 2 GB */
+                DISPLAYLEVEL(3, "Stream followed by undecodable data ");
+                if (position != -1L)
+                    DISPLAYLEVEL(3, "at position %i ", (int)position);
+                DISPLAYLEVEL(3, "\n");
+            }
+            break;
+        }
+      if(result != LZ4IO_LZ4F_OK){
+        break;
+      }
+      cfinfo->frameSummary = frameInfo; }
```

Comment:
Try to preserve a neat nesting level.
I'm sometimes lost, with these extra brackets.

Reply:
Improved formatting should be easier to follow now.